app/webpacker/components/Panel/pages/RunValidatorsPage/CompetitionRangeSelector.jsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import { Form } from 'semantic-ui-react';
+import { QueryClient, useQuery } from '@tanstack/react-query';
+import UtcDatePicker from '../../../wca/UtcDatePicker';
+import getCompetitionCount from './api/getCompetitionCount';
+import Loading from '../../../Requests/Loading';
+import Errored from '../../../Requests/Errored';
+
+const RUN_VALIDATORS_QUERY_CLIENT = new QueryClient();
+
+export default function CompetitionRangeSelector({ range, setRange }) {
+  const bothDatesAreSelected = Boolean(range?.startDate && range?.endDate);
+
+  const {
+    data: competitionCount, isLoading, isError,
+  } = useQuery({
+    queryKey: ['competitionCountInRange', range?.startDate, range?.endDate],
+    queryFn: () => getCompetitionCount(range?.startDate, range?.endDate),
+    enabled: bothDatesAreSelected,
+  }, RUN_VALIDATORS_QUERY_CLIENT);
+
+  return (
+    <>
+      <Form.Field
+        label="Start Date"
+        control={UtcDatePicker}
+        showYearDropdown
+        dateFormatOverride="yyyy-MM-dd"
+        dropdownMode="select"
+        isoDate={range?.startDate}
+        onChange={(date) => {
+          setRange({
+            ...range,
+            startDate: date,
+          });
```

Comment:
See my point about passing the values separately

Reply:
Changed.